From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: reintroduce strtobool
Patch-mainline: never, kabi
References: kabi

In 4.4.28, commit e0d61779d3fe56bf139c6e4c8381d9dd3eddc1f0 (lib: move
strtobool() to kstrtobool()), upstream commit
ef951599074ba4fad2d0efa0a977129b41e6d203 made an inline from strtobool
which was previously exported.

Reintroduce the function, just as a wrapper to a newly added
kstrtobool.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/string.h |    5 +----
 lib/string.c           |    6 ++++++
 2 files changed, 7 insertions(+), 4 deletions(-)

--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -127,11 +127,8 @@ extern char **argv_split(gfp_t gfp, cons
 extern void argv_free(char **argv);
 
 extern bool sysfs_streq(const char *s1, const char *s2);
+extern int strtobool(const char *s, bool *res);
 extern int kstrtobool(const char *s, bool *res);
-static inline int strtobool(const char *s, bool *res)
-{
-	return kstrtobool(s, res);
-}
 
 #ifdef CONFIG_BINARY_PRINTF
 int vbin_printf(u32 *bin_buf, size_t size, const char *fmt, va_list args);
--- a/lib/string.c
+++ b/lib/string.c
@@ -630,6 +630,12 @@ bool sysfs_streq(const char *s1, const c
 }
 EXPORT_SYMBOL(sysfs_streq);
 
+int strtobool(const char *s, bool *res)
+{
+	return kstrtobool(s, res);
+}
+EXPORT_SYMBOL(strtobool);
+
 #ifndef __HAVE_ARCH_MEMSET
 /**
  * memset - Fill a region of memory with the given value
